config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/FolderEntityBuilder.java

```diff
@@ -63,21 +64,34 @@ private Entity buildFolderEntity(Folder folder, String id, String parentFolderId
         return entity;
     }
 
-    private List<Entity> buildEntities(Map<String, ?> entities, BundleType bundleType, Document document) {
+    private List<Entity> buildEntities(Map<String, Folder> entities, BundleType bundleType, Document document, String targetFolder) {
         // no folder has to be added to environment bundle
-        if (entities.isEmpty() || bundleType == ENVIRONMENT) {
+        if ((entities.isEmpty() && StringUtils.isBlank(targetFolder)) || bundleType == ENVIRONMENT) {
             return Collections.emptyList();
         }
         Map<Folder, Collection<Folder>> folderChildrenMap = new HashMap<>();
-
-        entities.values().forEach(folder -> addFolder((Folder)folder, folderChildrenMap));
-
-        Folder rootFolder = (Folder)entities.get("");
+        Folder rootFolder = entities.remove("");
         if (rootFolder == null) {
             throw new EntityBuilderException("Could not locate root folder.");
         }
-        rootFolder.setId(Folder.ROOT_FOLDER_ID);
-        rootFolder.setName(Folder.ROOT_FOLDER_NAME);
+        String targetFolderPath = null;
+        if (StringUtils.isNotBlank(targetFolder)) {
+            Folder bundleTargetFolder = rootFolder;
+            targetFolderPath =  targetFolder + "/";
+            bundleTargetFolder.setId(idGenerator.generate());
+            bundleTargetFolder.setName(targetFolder);
+            bundleTargetFolder.setParentFolder(Folder.ROOT_FOLDER);
+            bundleTargetFolder.setPath(targetFolderPath);
+            entities.put(targetFolderPath , bundleTargetFolder);
+            rootFolder = Folder.ROOT_FOLDER;
+            entities.put("", rootFolder);
```

Comment:
Can targetFolder be with multiple levels like `root/folder1/folder2/` - in this case I need to deploy my entities in `folder2`, is this supported?

Reply:
targetFolder is just name of the folder its not path